src/main/java/io/anserini/index/IndexCollection.java

```diff
@@ -152,31 +182,36 @@ public void run() {
         while (iter.hasNext()) {
           SourceDocument d = iter.next();
           if (d == null) {
-            LOG.info("Null document encountered");
+            counters.unindexed.incrementAndGet();
```

Comment:
So I think this is a problem: 
Currently, if we encounter null document we do not know if it is EOF or the real null document (b/c of errors or sth). This is mainly due to how `Collection` read the documents. 
I think we should remove this for now and revisit it in #254